athena-oracle/src/main/java/com/amazonaws/athena/connectors/oracle/OracleJdbcConnectionFactory.java

```diff
@@ -60,22 +70,52 @@ public Connection getConnection(final JdbcCredentialProvider jdbcCredentialProvi
     {
         try {
             final String derivedJdbcString;
-            if (null != jdbcCredentialProvider) {
-                Matcher secretMatcher = SECRET_NAME_PATTERN.matcher(databaseConnectionConfig.getJdbcConnectionString());
-                final String secretReplacement = String.format("%s/%s", jdbcCredentialProvider.getCredential().getUser(), jdbcCredentialProvider.getCredential().getPassword());
-                derivedJdbcString = secretMatcher.replaceAll(Matcher.quoteReplacement(secretReplacement));
+            final Map<String, String> envVariables = System.getenv();
+
+            if (OracleConstants.YES.equalsIgnoreCase(envVariables.get("ssl"))) {
+                derivedJdbcString = databaseConnectionConfig.getJdbcConnectionString();
+                Pattern connStringPattern = Pattern.compile("jdbc:oracle:thin:@\\((?i)description=\\(address=\\(protocol=tcps\\)\\(host=[a-zA-Z0-9-.]+\\)" +
+                        "\\(port=([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])\\)\\)" +
+                        "\\(connect_data=\\(sid=[a-zA-Z_]+\\)\\)\\(security=\\(ssl_server_cert_dn=\\\"[=a-zA-Z,0-9-.,]+\\\"\\)\\)\\)");
```

Comment:
This should be a `static final SSLConnStringPattern` at the class level.

I know this is just in a lambda so it will be compiled every time the lambda runs anyway, but its just a code quality thing, where we want things that are logically `static final` as a `static final` into the class level.